inc/class-visitor.php

```diff
@@ -283,13 +286,14 @@ public function log_vote( $comment_id, $action ) {
 	public function unlog_vote( $comment_id ) {
 
 		$comments_voted_on = $this->retrieve_logged_votes();
-		$logged_votes      = $this->retrieve_logged_votes();
-
-		unset( $logged_votes[ 'comment_id_' . $comment_id ] );
+		unset( $comments_voted_on[ $comment_id ] );
 
-		if ( ! empty( $comments_voted_on ) ) {
+		if ( !empty( $comments_voted_on ) ) {
 			update_user_option( $this->get_id(), 'hmn_comments_voted_on', $comments_voted_on );
+		} else {
+			delete_user_option( $this->get_id(), 'hmn_comments_voted_on' );
```

Comment:
This could just be updated to an empty array, the `! empty()` check actually seems unnecessary looking at this. Good catch :)